routes/dist_tags.js

```diff
@@ -1,54 +1,41 @@
 'use strict'
 
-const co = require('co')
 const r = require('koa-router')()
 const npm = require('../lib/npm')
 const config = require('../config')
-const job = require('../lib/job')
-const _ = require('lodash')
 const parse = require('co-body')
 const middleware = require('../middleware')
 
-let updateDistTags = job(co.wrap(function * (name, tags) {
-  let current = yield getStorageTags(name)
-  if (!_.isEqual(current, tags)) yield config.storage.put(`dist-tags/${name}`, tags)
-}))
-
-function getStorageTags (name) {
-  return config.storage.getJSON(`dist-tags/${name}`)
-}
+let getPackage = name => config.storage.getJSON(`packages/${name}`)
 
 r.get('/-/package/:name/dist-tags', function * () {
   let {name} = this.params
-  try {
-    let tags = yield npm.getDistTags(name)
-    this.body = tags
-    updateDistTags(name, tags)
-  } catch (err) {
-    let tags = yield getStorageTags(name)
-    if (!tags) throw err
-    this.body = tags
+  let pkg = yield npm.get(name)
+  if (pkg !== 404) this.body = pkg['dist-tags']
```

Comment:
@dickeyxxx realized that this will not hit 304 because we do not pass the etag parameter, going to ship now, with the caveat that we should not map errors to 404s soon